### DISCUSSION TITLE
Does Nautilus offer market, and stop market, orders with protection?

### DISCUSSION THREAD

**Discussion by eliahburns** [nautechsystems/nautilus_trader#1079] [Q&A] [answered]
Hi, I'm just getting up to speed on Nautilus and trying to verify whether I can use it with my strategy. I was looking through the order types, and I haven't spotted a way to configure market (or stop market) orders with protection like [CME offers](https://www.cmegroup.com/education/protection-functionality-for-market-and-stop-orders.html). 

Have I missed this, and it exists? Or is it planned to be implemented? 

Thanks in Advance!

**Comment by cjdsellers** [comment]
Hi @eliahburns 

This is a great question, I've also noticed this would be needed to provide that functionality for execution with CME. Consider this to be on the nearer term roadmap as its really just a case of adding a param and field, I'll have to review their spec though.

**Reply by eliahburns** [reply]
You rock @cjdsellers 

I’m open to contributing as well if you're short handed and would want to point me in the general direction of what is needed. I know interactive brokers offers this [functionality](https://www.interactivebrokers.com/en/trading/orders/stop-with-protection.php) through their API as well--I'm not sure whether that integration would need to be explicitly modified without looking at the code more.

**Comment by cjdsellers** [comment, marked as answer]
Now added to the backlog https://github.com/nautechsystems/nautilus_trader/issues/1082.
